### ISSUE TITLE
[Story] U-06: Message from Bob is decorated with "Verified identity has changed"

### ISSUE THREAD

**Issue by andybalaam** [vector-im/element-meta#2523]
### Description

As Alice
When I am receiving a message from Bob
And we have verified each other with Bob in the past
But Bob's identity has changed after this
Then I want to see Bob's message decorated with "Verified identity has changed"
So that I can be aware that this may be malicious but I can continue conversation to determine how to proceed

As Alice
When I have re-verified Bob or withdrawn verification after Bob's identity changed
Then I want Bob's past messages no longer be decorated with "Verified identity has changed"
So that the current state of the verification is reflected.

Note: currently a message from a verified-identity that has changed, is not decrypted but instead a static placeholder text "Verified identity has changed" is shown.

Current state:

<img width="356" alt="Image" src="https://github.com/user-attachments/assets/17d92251-a23c-466c-9029-2d6e487741c6" />

Desired state:

<img width="356" height="117" alt="Image" src="https://github.com/user-attachments/assets/4995bc53-5a70-4491-9088-3e59db9061ab" />

**Designs**
* [iOS](https://www.figma.com/design/0oUTCbYhROEDprSp7bdWhx/Identity-Changes?node-id=895-9077&t=u3djDFGxksLmI0Qt-4) | [Android](https://www.figma.com/design/0oUTCbYhROEDprSp7bdWhx/Identity-Changes?node-id=224-33098&t=u3djDFGxksLmI0Qt-4) | [Web](https://www.figma.com/design/0oUTCbYhROEDprSp7bdWhx/Identity-Changes?node-id=220-47274&t=u3djDFGxksLmI0Qt-4)

### Acceptance criteria

- TBD

### Size estimate

None

### Dependencies

- None

### Out of scope

- Nothing

### Open questions

```[tasklist]
### Questions
```

### Subtasks

```[tasklist]
### Android
- [ ] Todo
```

```[tasklist]
### iOS
- [ ] Todo
```

```[tasklist]
### Rust
```

```[tasklist]
### Web
- [ ] Todo
```

**Comment by mxandreas** [comment]
Regarding push notifications, it was decided that we show the push notification but would not show the message content. Instead, we warn that the verified identity has changed. @americanrefugee also added an example of such push notification in the design but I can't find it anymore.

**Comment by americanrefugee** [comment]
Here is the design for the push notification: https://www.figma.com/design/0oUTCbYhROEDprSp7bdWhx/Identity-Changes?node-id=443-12319&t=IIEWvRUWrjnXNS9d-4

**Comment by uhoreg** [comment]
@americanrefugee do we have designs for how this should look like in web for:

1. message preview in the the room sidebar
2. thread preview in the timeline

(do the mobile apps also have an equivalent to message preview in their room list?)

**Comment by richvdh** [comment]
Some discussion [here](https://github.com/element-hq/crypto-internal/issues/362#issuecomment-2357502745) about how this might be implemented in EWR

**Comment by americanrefugee** [comment]
> @americanrefugee do we have designs for how this should look like in web for:
> 
> 1. message preview in the the room sidebar
> 2. thread preview in the timeline
> 
> (do the mobile apps also have an equivalent to message preview in their room list?)

@uhoreg Is this what you need?
- [Threaded message in timeline and Threads list](https://www.figma.com/design/0oUTCbYhROEDprSp7bdWhx/Identity-Changes?node-id=613-19753&t=v8GKPR2tz14yqF42-4)
- [Thread message](https://www.figma.com/design/0oUTCbYhROEDprSp7bdWhx/Identity-Changes?node-id=613-30438&t=v8GKPR2tz14yqF42-4)

FYI @mxandreas

**Comment by uhoreg** [comment]
@americanrefugee Yes, for threads.  There's also the message preview in the room sidebar (on the left)

**Comment by americanrefugee** [comment]
> @americanrefugee Yes, for threads. There's also the message preview in the room sidebar (on the left)

Is there? I don't see message previews in the Room List...  

However, in general I think we should:
- NOT show an indicator for a room in the room list
- [But DO show an indicator for a DM](https://www.figma.com/design/0oUTCbYhROEDprSp7bdWhx/Identity-Changes?node-id=613-19753&t=UJIs5sxxCA1XQm0b-4) in the list room if the user was previously verified

**Comment by uhoreg** [comment]
> Is there? I don't see message previews in the Room List...

Message preview in the room list is an optional thing that can be enabled by clicking on the "..." button that appears when you hover over the room type header (e.g. "People", "Rooms") in the room list.

![image](https://github.com/user-attachments/assets/60ba8cb1-e535-4684-9e0d-4cd612474b4a)

Another thing is, should we do anything with incoming message notifications?

**Comment by mxandreas** [comment]
> NOT show an indicator for a room in the room list
> [But DO show an indicator for a DM](https://www.figma.com/design/0oUTCbYhROEDprSp7bdWhx/Identity-Changes?node-id=613-19753&t=UJIs5sxxCA1XQm0b-4) in the list room if the user was previously verified

@americanrefugee What would be your reasoning of this?

Also, just in case a reminder that we have 2 related but not tightly coupled situations:
* a) Bob's verified identity has changed.
* b) A message was received from Bob (assumes a)). 

So far, we have applied independent measures (decorations) to those and in this ticket we were focused on b).

However, if I think about the room list, then it will be probably hard to differentiate in there between a) and b); at least not so that the user understands. Thus we need to think we show something in there already in case of a) or only when b) happens.

> Another thing is, should we do anything with incoming message notifications?

@uhoreg which notifications do you mean? I think this is where the topic started from; and Aaron provided the designs in [here](https://github.com/element-hq/element-meta/issues/2523#issuecomment-2358203269).

**Comment by americanrefugee** [comment]
> @americanrefugee What would be your reasoning of this?

What we don't want to do is scare the user or clutter the interface with decorations. Imagine that the unverified person replied to 3 rooms. Your room list would look like there are multiple errors, but that's not true. Or else the error appears and then disappears when the next message comes in from a normal user. Which would also be confusing.

But if we decorate the DM from the unverified user, then you know that something is wrong with that user and not all the rooms.

Plus, you will see sufficient warning once in a room to know to take action. Before that, it isn't critical that you take action because everything else is working fine.

> Also, just in case a reminder that we have 2 related but not tightly coupled situations:
> 
> * a) Bob's verified identity has changed.
> * b) A message was received from Bob (assumes a)).
> 
> So far, we have applied independent measures (decorations) to those and in this ticket we were focused on b).
> 
> However, if I think about the room list, then it will be probably hard to differentiate in there between a) and b); at least not so that the user understands. Thus we need to think we show something in there already in case of a) or only when b) happens.

Not following you here... But still, whatever room that has that message will tell you what's going on.

> > Another thing is, should we do anything with incoming message notifications?

Yes, [there is new text for notifications](https://www.figma.com/design/0oUTCbYhROEDprSp7bdWhx/Identity-Changes?node-id=443-12319&t=hZdQHDFPv2Fa3Zj2-4).

**Comment by mxandreas** [comment]
We were recently reviewing the current placeholder texts for messages that we do not decrypt - to make the icons and texts and formatting consistent across the platform. The outcomes in terms of what to fix were captured in here: https://github.com/element-hq/element-meta/issues/3082 - which are overall independent of this story.

However when we discussed the formatting of the placeholder texts, it opened a wider discussion of which kind of different patterns we have, and it seemed like it is worth having another pattern (e.g. red icon with a text on the message bubble). Instead we should probably reuse the red dot within the message bubble that we already have. This should be to be confirmed before we start working on this.
